libcloud/compute/drivers/openstack.py

```diff
@@ -335,6 +335,34 @@ def ex_hard_reboot_node(self, node):
         """
         return self._reboot_node(node, reboot_type='HARD')
 
+    def ex_list_subnet(self, network_id=None):
+        """
+        List all subnet resources
+        :param        network_id:   filter for subnets for this network ID
+        :type         network_id:   ``str``
+        :rtype:       ``list`` of :class:``OpenStackSubnet``
+        """
+        uri = '/subnets'
+        resp = self.connection.request(uri, method='GET')
+        if resp.status == httplib.NOT_FOUND:
+            return None
+        return self._to_subnets(resp.object)
+
+
+class OpenStackSubnet(object):
```

Comment:
To make it easier to eventually switch to #345, can you please make sure this class looks more similar to the one here https://github.com/apache/libcloud/pull/345/files#diff-4c75f73aaa54353918c9fc5a4fc083e5R102 https://github.com/apache/libcloud/pull/345/files#diff-5cc9e28889b2f8491dcdbc2114a213b0R65?